src/server/routes/add_label.ts

```diff
@@ -0,0 +1,46 @@
+import { Request, Response } from "express";
+
+import { endWithStatusAndBody } from "../helpers/http";
+import { getBqConnection } from "../helpers/bigquery";
+
+type addLabelPayload = {
+  report_uuid?: string;
+  label?: string;
+};
+
+export default async function handleAddLabel(req: Request, res: Response) {
+  const payload = req.body as addLabelPayload;
+  if (!payload.report_uuid || !payload.label) {
+    return endWithStatusAndBody(res, 400, "Missing report_uuid or label");
+  }
+
+  const bq = getBqConnection();
+  try {
+    const queries = [
+      bq.query({
+        query: `INSERT INTO webcompat_user_reports.labels (report_uuid, label, created_at, is_ml)
+        VALUES (?, ?, CURRENT_DATETIME(), false);
+      `,
+        params: [payload.report_uuid, payload.label],
+      }),
+    ];
+
+    if (payload.label == "invalid") {
+      queries.push(
+        bq.query({
+          query: `
+            INSERT INTO webcompat_user_reports.report_actions (report_uuid, type, created_at)
+            VALUES (?, "mark-invalid", CURRENT_DATETIME())
+        `,
+          params: [payload.report_uuid],
+        }),
+      );
+    }
```

Comment:
Since the main User Reports view doesn't show reports that have any actions I only have `mark-invalid` action here. So in case the report labelled as valid, it will still be displayed. 

But to make it consistent, I can probably change the user report query in `fetchUserReports` to exclude `mark-valid`, so it'll be something like:

```
          CASE WHEN EXISTS (SELECT 1 FROM webcompat_user_reports.report_actions WHERE report_actions.report_uuid = reports.document_id and report_actions.type != "mark-valid")
            THEN true ELSE false END AS has_actions
```

Reply:
This makes sense, yeah. +1 to changing the main report query - I don't think it matters too much since I don't think we'd be spending a lot of time manually triaging the classifications, but this could be confusing otherwise.